driver-core/src/test/java/com/datastax/driver/core/ProtocolVersionRenegotiationTest.java

```diff
@@ -0,0 +1,136 @@
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.exceptions.UnsupportedProtocolVersionException;
+import org.testng.annotations.Test;
+
+import static com.datastax.driver.core.ProtocolVersion.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@CreateCCM(CreateCCM.TestMode.PER_METHOD)
+public class ProtocolVersionRenegotiationTest extends CCMTestsSupport {
+
+    /**
+     * @jira_ticket JAVA-1367
+     */
+    @Test(groups = "short")
+    @CCMConfig(version = "3.6", createCluster = false)
+    public void should_fail_when_renegotiation_disabled_3_6() throws Exception {
+        UnsupportedProtocolVersionException e = connectWithUnsupportedVersion(V1);
+        assertThat(e.getUnsupportedVersion()).isEqualTo(V1);
+        // pre-CASSANDRA-11464: server replies with its own version
+        assertThat(e.getServerVersion()).isEqualTo(V4);
+    }
+
+    /**
+     * @jira_ticket JAVA-1367
+     */
+    @Test(groups = "short")
+    @CCMConfig(version = "3.10", createCluster = false)
+    public void should_fail_when_renegotiation_disabled_3_10() throws Exception {
+        UnsupportedProtocolVersionException e = connectWithUnsupportedVersion(V1);
+        assertThat(e.getUnsupportedVersion()).isEqualTo(V1);
+        // post-CASSANDRA-11464: server replies with client's version
+        assertThat(e.getServerVersion()).isEqualTo(V1);
+    }
+
+    /**
+     * @jira_ticket JAVA-1367
+     */
+    @Test(groups = "short")
+    @CCMConfig(version = "2.1.16", createCluster = false)
+    public void should_fail_when_initial_version_is_beta() throws Exception {
+        UnsupportedProtocolVersionException e = connectWithUnsupportedBetaVersion();
+        assertThat(e.getUnsupportedVersion()).isEqualTo(V5);
+        assertThat(e.getServerVersion()).isEqualTo(V3);
+    }
+
+    /**
+     * @jira_ticket JAVA-1367
+     */
+    @Test(groups = "short")
+    @CCMConfig(version = "1.2.19", createCluster = false)
+    public void should_renegotiate_when_renegotiation_enabled_and_initial_version_too_high_1_2_19() throws Exception {
+        Cluster cluster = connectWithoutVersion();
+        assertThat(actualProtocolVersion(cluster)).isEqualTo(V1);
+    }
+
+    /**
+     * @jira_ticket JAVA-1367
+     */
+    @Test(groups = "short")
+    @CCMConfig(version = "2.0.17", createCluster = false)
+    public void should_renegotiate_when_renegotiation_enabled_and_initial_version_too_high_2_0_17() throws Exception {
```

Comment:
`@CCMConfig(version = "XXX")` forces a specific version instead of skipping the test if the version doesn't match the default one. I think it is required sometimes too, as tests like these could fail if run against any other version.